From c4bb60046de9ffdd1cf07cfcdba201df06b194dd Mon Sep 17 00:00:00 2001
From: Maarten Coene <maartenc@apache.org>
Date: Tue, 13 Dec 2016 00:48:36 +0100
Subject: [PATCH] Throw an IllegalStateException when retrieving the
 resolutionCacheRoot on the DefaultResolutionCacheManager if the basedir (or
 IvySettings) is not set (IVY-1482)

---
 doc/release-notes.html                                    | 1 +
 .../ivy/core/cache/DefaultResolutionCacheManager.java     | 8 +++++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/doc/release-notes.html b/doc/release-notes.html
index 085e49463..688d72247 100644
--- a/doc/release-notes.html
+++ b/doc/release-notes.html
@@ -68,6 +68,7 @@ <h2>List of Changes in this Release</h2>
 - FIX: Dependencies failed using branch attribute (and extra attributes) (IVY-1141) (Thanks to Stephen Haberman)
 - FIX: useCacheOnly should allow lookup of changing dependencies in cache (IVY-1515) (Thanks to Ilya)
 
+- IMPROVEMENT: Throw an IllegalStateException when retrieving the resolutionCacheRoot on the DefaultResolutionCacheManager if the basedir (or IvySettings) is not set (IVY-1482)
 - IMPROVEMENT: Optimization: limit the revision numbers scanned if revision prefix is specified (Thanks to Ernestas Vaiciukevi&#269;ius)
 - IMPROVEMENT: Update bouncycastle to 1.52 (IVY-1521) (Thanks to Michal Srb)
 
diff --git a/src/java/org/apache/ivy/core/cache/DefaultResolutionCacheManager.java b/src/java/org/apache/ivy/core/cache/DefaultResolutionCacheManager.java
index d6bc2d72e..9c4214416 100644
--- a/src/java/org/apache/ivy/core/cache/DefaultResolutionCacheManager.java
+++ b/src/java/org/apache/ivy/core/cache/DefaultResolutionCacheManager.java
@@ -74,6 +74,12 @@ public void setSettings(IvySettings settings) {
     }
 
     public File getResolutionCacheRoot() {
+        if (basedir == null) {
+            if (settings == null) {
+                throw new IllegalStateException("The 'basedir' or 'IvySettings' has not been set on the ResolutionCacheManager");
+            }
+            basedir = settings.getDefaultResolutionCacheBasedir();
+        }
         return basedir;
     }
 
@@ -213,7 +219,7 @@ public String toString() {
     }
 
     public void clean() {
-        FileUtil.forceDelete(getBasedir());
+        FileUtil.forceDelete(getResolutionCacheRoot());
     }
 
     private static class CacheParserSettings implements ParserSettings {
